python/src/trezorlib/transport/__init__.py

```diff
@@ -79,6 +79,9 @@ def read(self) -> MessagePayload:
     def write(self, message_type: int, message_data: bytes) -> None:
         raise NotImplementedError
 
+    def find_debug(self) -> "Transport":
```

Comment:
we should be using something like self-types here (and also in the methods below -- i didn't do it originally because mypy was too weak, but maybe pyright can manage

the idea is, `find_debug(self: T) -> T` `enumerate(cls: Type[T]) -> Iterable[T]`, etc.

(we unfortunately don't have the Self type yet)

Reply:
I tried the below-mentioned without success:
```python
BT = TypeVar("BT", bound="BridgeTransport")

def find_debug(self: "BT") -> "BT":
        if not self.device.get("debug"):
            raise TransportException("Debug device not available")
        return BridgeTransport(self.device, self.legacy, debug=True)
```

error is `Expression of type "BridgeTransport" cannot be assigned to return type "BT@find_debug"`

Reply:
Not specifying the type of `Transport` also does not help (`BT = TypeVar("BT")`):
```
  /home/jmusil/trezor-firmware/python/src/trezorlib/transport/bridge.py:131:21 - error: Cannot access member "device" for type "object*"
    Member "device" is unknown (reportGeneralTypeIssues)
  /home/jmusil/trezor-firmware/python/src/trezorlib/transport/bridge.py:133:37 - error: Cannot access member "device" for type "object*"
    Member "device" is unknown (reportGeneralTypeIssues)
  /home/jmusil/trezor-firmware/python/src/trezorlib/transport/bridge.py:133:50 - error: Cannot access member "legacy" for type "object*"
  /home/jmusil/trezor-firmware/python/src/trezorlib/transport/bridge.py:133:16 - error: Expression of type "BridgeTransport" cannot be assigned to return type "BT@find_debug"
    Type "BridgeTransport" cannot be assigned to type "BT@find_debug" (reportGeneralTypeIssues)
  /home/jmusil/trezor-firmware/python/src/trezorlib/transport/bridge.py:146:20 - error: Expression of type "list[BridgeTransport]" cannot be assigned to return type "Iterable[BT@enumerate]"
    TypeVar "_T_co@Iterable" is covariant
      Type "BridgeTransport" cannot be assigned to type "BT@enumerate" (reportGeneralTypeIssues)
```

Reply:
On `Transport` itself, it works fine - question is, if it helps, when not applied on the children classes

Strict mode has found 1702 errors, so disabling/ignoring/resolving those could be a big quest, but I am up to it

Reply:
> if it helps, when not applied on the children classes

it very much does.
see `UdpTransport.find_by_path()`, you can remove the whole `cast()` weirdness and "mypy can't do this" comment if the parent class is correctly typed

Reply:
(this is btw reported by "useless cast" checker)

Reply:
Done in 263d7b5d6 and f32c090ff (fixup for that UDP removal)